packages/flutter/lib/src/painting/binding.dart

```diff
@@ -143,30 +143,15 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
     final String type = message['type'] as String;
     switch (type) {
       case 'fontsChange':
-        _systemFonts.notifyListeners();
+        _systemFonts._notify();
         break;
     }
     return;
   }
 }
 
-class _SystemFontsNotifier extends Listenable {
-  final Set<VoidCallback> _systemFontsCallbacks = <VoidCallback>{};
-
-  void notifyListeners () {
-    for (final VoidCallback callback in _systemFontsCallbacks) {
-      callback();
-    }
-  }
-
-  @override
-  void addListener(VoidCallback listener) {
-    _systemFontsCallbacks.add(listener);
-  }
-  @override
-  void removeListener(VoidCallback listener) {
-    _systemFontsCallbacks.remove(listener);
-  }
+class _SystemFontsNotifier extends ChangeNotifier {
```

Comment:
the changeNotifier uses list and the text renderobject add/remove itself in attach and detach, it may depends on how you test it. I am not sure, but is it possible to override the hashcode method of the closure?

Reply:
Hmm. Attach and detach might not be the right place for this - now that RO has dispose, perhaps we should move it to creation and disposal?

Reply:
Can you explain more? if detached RO doesn't need to be notified about font change

Reply:
Ahh ok. I'm running a benchmark like this: https://gist.github.com/dnfield/51b8f5ee3a0439b47d2841fc688d95c6, where this change is showing significant improvement. 

I'll try this change out in a larger "real application" benchmark as well though.

Reply:
In flutter_gallery__transitions_perf on both a Android Go and a Pixel 4 device, this all seems to be lost in noise. It improve the microbenchmark but not the real application benchmark.

Given that, I'd be fine with just waiting on a fix for hashing closures in the upstream Dart SDK.